ddl/ddl_api.go

```diff
@@ -1323,25 +1324,25 @@ func (d *ddl) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (err e
 
 	err = d.doDDLJob(ctx, job)
 	if err == nil {
-		var preSplitAndScatter func()
 		// do pre-split and scatter.
-		if tbInfo.ShardRowIDBits > 0 && tbInfo.PreSplitRegions > 0 {
-			preSplitAndScatter = func() {
-				preSplitTableShardRowIDBitsRegion(d.store, tbInfo, ctx.GetSessionVars().WaitSplitRegionFinish)
+		sp, ok := d.store.(kv.SplitableStore)
+		if ok && atomic.LoadUint32(&EnableSplitTableRegion) != 0 {
+			var preSplit func()
+			val, err := variable.GetGlobalSystemVar(ctx.GetSessionVars(), variable.TiDBScatterRegion)
+			if err != nil {
+				logutil.BgLogger().Warn("[ddl] pre-split region failed", zap.Error(err))
 			}
-		} else if atomic.LoadUint32(&EnableSplitTableRegion) != 0 {
+			scatterRegion := variable.TiDBOptOn(val)
```

Comment:
if the `variable.GetGlobalSystemVar` return err, scatterRegion should be false?

Reply:
Yes, I think so.